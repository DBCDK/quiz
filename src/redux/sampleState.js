import Immutable from 'immutable';
import {quizData} from '../quizData.js';

export const sampleState = Immutable.fromJS({
  admin: {
    // edit a single screen if currentScreen is set
    currentScreen: undefined
  },
  widget: {
    // will be set to true, when all data is loaded
    initialised: false,
    // non-zero if loading, increases with each load operation, and decreases when done
    loading: 0,
    currentScreen: undefined
  },
  quizState: {},

  // data for the current Quiz
  quiz: undefined,

  searchQuery: {
    query: '',
    ownOnly: true
  },

  searchResults: [],

  storage: {
    // storage user id, initialised on start
    user: undefined,

    // storage type uuids, initialised on start
    quizType: undefined,
    imageType: undefined,

    // mirror of data in storage, used for to find local changes to sync to storage
    quiz: {},
    page: {}
  }
});
export const newQuiz = sampleState.set('quiz', Immutable.fromJS(quizData()));
