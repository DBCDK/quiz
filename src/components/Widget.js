import React, {Component} from 'react';
import {ownQuizzes, loading} from '../redux/selectors';

class Widget extends Component {
  render() {
    return (
      <div>
        <h1>Quiz Widget here...</h1>
      </div>
    );
  }
}

export function mapStateToProps(state, ownProps) {
  return {
    loading: loading(state),
    ownQuizzes: ownQuizzes(state)
  };
}

export default Widget;
