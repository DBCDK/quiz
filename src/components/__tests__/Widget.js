import React from 'react';
import ReactDOM from 'react-dom';
import {Widget, mapStateToProps, mapDispatchToProps} from '../Widget';
import renderer from 'react-test-renderer';
import sampleState from '../../redux/sampleState';
import Immutable from 'immutable';

it('maps state to props', () => {
  expect(mapStateToProps(sampleState)).toMatchSnapshot();
});
it('maps dispatch to props', () => {
  expect(mapDispatchToProps(() => {})).toMatchSnapshot();
});
it('renders', () => {
  const tree = renderer.create(<Widget quizTitle="Titel for Quiz" />).toJSON();
  expect(tree).toMatchSnapshot();
});
it('renders page', () => {
  const page = Immutable.fromJS([
    {type: 'text', text: 'hello'},
    {
      type: 'buttonGroup',
      ui: [{type: 'button', text: 'but'}]
    },
    {type: 'image', image: 'some://url'}
  ]);
  const tree = renderer
    .create(
      <Widget
        quizTitle="Titel for Quiz"
        ui={page}
        vars={Immutable.fromJS({})}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
