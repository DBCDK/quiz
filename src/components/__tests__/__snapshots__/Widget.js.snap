// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps dispatch to props 1`] = `
Object {
  "onAction": [Function],
}
`;

exports[`maps state to props 1`] = `
<div>
  "Quiz screen missing or loading..."
</div>
`;

exports[`renders 1`] = `
Array [
  <header
    className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-8"
>
    <div
        className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
    >
        <h2
            className="MuiTypography-root-41 MuiTypography-title-47 MuiTypography-colorInherit-61"
        >
            Titel for Quiz
        </h2>
    </div>
</header>,
  <center
    style={
        Object {
            "background": undefined,
            "backgroundSize": "cover",
          }
    }
>
    <div
        className="MuiGrid-container-67 MuiGrid-spacing-xs-16-90"
    >
        <div
            className="MuiGrid-item-68 MuiGrid-grid-xs-12-107"
        />
    </div>
</center>,
]
`;

exports[`renders page 1`] = `
Array [
  <header
    className="MuiPaper-root-173 MuiPaper-elevation4-179 MuiAppBar-root-164 MuiAppBar-positionStatic-168 MuiAppBar-colorPrimary-171"
>
    <div
        className="MuiToolbar-root-200 MuiToolbar-regular-202 MuiToolbar-gutters-201"
    >
        <h2
            className="MuiTypography-root-204 MuiTypography-title-210 MuiTypography-colorInherit-224"
        >
            Titel for Quiz
        </h2>
    </div>
</header>,
  <center
    style={
        Object {
            "background": undefined,
            "backgroundSize": "cover",
          }
    }
>
    <div
        className="MuiGrid-container-230 MuiGrid-spacing-xs-16-253"
    >
        <div
            className="MuiGrid-item-231 MuiGrid-grid-xs-12-270"
        />
        <div
            className="MuiGrid-item-231 MuiGrid-grid-xs-12-270"
        >
            <div
                className="MuiTypography-root-204 MuiTypography-body1-213"
            >
                <div
                    dangerouslySetInnerHTML={
                        Object {
                            "__html": "<p>hello</p>
                        ",
                          }
                    }
                />
            </div>
        </div>
        <div
            className="MuiGrid-item-231 MuiGrid-grid-xs-12-270"
        >
            <div
                className="MuiGrid-container-230 MuiGrid-spacing-xs-16-253"
            >
                <div
                    className="MuiGrid-item-231 MuiGrid-grid-xs-6-264"
                >
                    <button
                        className="MuiButtonBase-root-353 MuiButton-root-327 MuiButton-contained-338 MuiButton-raised-341 MuiButton-fullWidth-352"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex="0"
                        type="button"
                    >
                        <span
                            className="MuiButton-label-328"
                        >
                            but
                        </span>
                        <span
                            className="MuiTouchRipple-root-356"
                        />
                    </button>
                </div>
            </div>
        </div>
        <div
            className="MuiGrid-item-231 MuiGrid-grid-xs-12-270"
        >
            <center>
                <img
                    alt=""
                    className={undefined}
                    src="some://url"
                />
            </center>
        </div>
    </div>
</center>,
]
`;
